OpenSim/Framework/Communications/UserProfileManager.cs

```diff
@@ -240,6 +240,12 @@ public UserProfileData GetUserProfile(UUID uuid, bool forceRefresh)
                 profile = TryGetUserProfile(uuid, false);
                 if (profile != null)
                 {
+                    if (profile.CurrentAgent == null)
```

Comment:
The AgentData isn't the profile. But even if it was, this fix is not any kind of refresh, the *problem* is that the AgentData is null.  The last logoff location is in that object, and it has never been *loaded* in the first place.  For the case of refreshes, we tie the refresh of that to main profile refreshes, but again this isn't a refresh, it's null.